hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/GrpcOzoneManagerServer.java

```diff
@@ -46,24 +58,42 @@ public GrpcOzoneManagerServer(OzoneConfiguration config,
                                 OzoneManagerProtocolServerSideTranslatorPB
                                     omTranslator,
                                 OzoneDelegationTokenSecretManager
-                                    delegationTokenMgr) {
+                                    delegationTokenMgr,
+                                CertificateClient caClient) {
     this.port = config.getObject(
         GrpcOzoneManagerServerConfig.class).
         getPort();
     init(omTranslator,
         delegationTokenMgr,
-        config);
+        config,
+        caClient);
   }
 
   public void init(OzoneManagerProtocolServerSideTranslatorPB omTranslator,
                    OzoneDelegationTokenSecretManager delegationTokenMgr,
-                   OzoneConfiguration omServerConfig) {
+                   OzoneConfiguration omServerConfig,
+                   CertificateClient caClient) {
     NettyServerBuilder nettyServerBuilder = NettyServerBuilder.forPort(port)
         .maxInboundMessageSize(OzoneConsts.OZONE_SCM_CHUNK_MAX_SIZE)
         .addService(new OzoneManagerServiceGrpc(omTranslator,
             delegationTokenMgr,
             omServerConfig));
 
+    SecurityConfig secConf = new SecurityConfig(omServerConfig);
+    if (secConf.isGrpcTlsEnabled()) {
```

Comment:
Is is possible that GRPC TLS is enabled but Ozone security is not. What would happen in that case?

Reply:
Normal configuration would be for _ozone.security.enabled_ **and** _hdds.grpc.tls.enabled_ set to _'true'_ as is the case for our docker secure cluster configurations.

In the event GRPC TLS is enabled but Ozone security is not, the certificate list (TLS authenication) used by the client (`GrpcOmTransport`) will not be available (set to null) and the client will not be able to authenticate the server certificate to establish a secure channel between client and server.  On the server side (`GrpcOzoneManagerServer`) receives a connection error.  The error is logged on the client side as `"x509Certicates empty"` and the client receives a 500 http response.  The log and errors should provide useful info to the admin to correct configuration errors.

Reply:
Yes, logging a separate error for misconfigured ozone.security while _grpc.tls.enabled_ is `true` for the s3g Grpc TLS channel is much better. 

I have modified the code for this case on both the s3g Grpc client and the s3g Grpc server.  For the client, `GrpcOmTransport`, should `grpc.tls.enabled` be set when _ozone.security_ = `false`, we log the error "...security not enabled when TLS specified.".  Also, we don't set the sslContextBuilder for tls and default to plaintext channel.  Similarly, for the server, `GrpcOzoneManagerServer`, should `grpc.tls.enabled` be set when _ozone.security_ = `false`, we log the error ("... not enabled when TLS..., using plaintext.") and default to a plaintext channel.

Reply:
Latest commit contains changes to handle ozone-site misconfigurations for TLS enabled s3g Grpc and ozone.security.enabled.